ledger-api/grpc-definitions/com/daml/ledger/api/v1/error.proto

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+// Rules on error reporting over the Ledger API
+// ********************************************
+//
+// 1. We use gRPC status codes to report synchronous success and failures
+//    coming from the underlying gRPC protocol or the API-level request processing.
```

Comment:
Do you have a definition of API-level? I think I know what you mean but I have trouble pinning it down precisely in a way that gets at the difference between the CommandService and the CommandSubmissionService.

Reply:
The idea is that API-level errors are those that our implementation of the gRPC endpoint produces, in contrast to errors coming from the underlying gRPC library or even some transport layer. It has nothing to do with command service vs. command submission service. The distinction is made only because we can't really influence how those libraries produce their errors.

Reply:
Thanks for the explanation, makes sense